geoapps/octree_creation/driver.py

```diff
@@ -67,21 +67,37 @@ def octree_from_params(params: OctreeParams):
             mesh_type="tree",
             depth_core=params.depth_core,
         )
+        minimum_level = max([1, treemesh.max_level - params.minimum_level + 1])
+        treemesh.refine(minimum_level, finalize=False)
 
         for label, value in params.free_parameter_dict.items():
-            if not isinstance(getattr(params, value["object"]), ObjectBase):
+            ref_entity = getattr(params, value["object"])
+            levels = utils.str2list(getattr(params, value["levels"]))
+            if not isinstance(ref_entity, ObjectBase):
                 continue
 
             print(f"Applying {label} on: {getattr(params, value['object']).name}")
 
-            treemesh = refine_tree_xyz(
-                treemesh,
-                getattr(params, value["object"]).vertices,
-                method=getattr(params, value["type"]),
-                octree_levels=utils.str2list(getattr(params, value["levels"])),
-                max_distance=getattr(params, value["distance"]),
-                finalize=False,
-            )
+            if isinstance(ref_entity, Curve):
+                locs = densify_curve(ref_entity, treemesh.h[0][0])
```

Comment:
This should be it's own private method with signature: treemesh = refine_tree_along_curve(treemesh, curve, levels).

Reply:
Yep, will be straightforward to re-implement the other refinement methods that way for GEOPY-1059